ec2utils/ec2utilsbase/lib/ec2utils/ec2preparer.py

```diff
@@ -0,0 +1,173 @@
+
+    # ---------------------------------------------------------------------
+    def _create_route_table(self):
+        response = self._connect().create_route_table(VpcId=self.vpc_id)
+        self.route_table_id = response['RouteTable']['RouteTableId']
+        self._connect().create_route(DestinationCidrBlock='0.0.0.0/0', GatewayId=self.internet_gateway_id, RouteTableId=self.route_table_id)
+        if self.verbose:
+            print("Created route table with id %s" % (self.route_table_id))
+
+    # ---------------------------------------------------------------------
+    def _create_security_group(self):
+        if self.verbose:
+            print('Creating temporary security group')
+        group_name = 'ec2uploadimg-%s' % (random.randint(1,100))
+        group_description = 'ec2uploadimg created %s' % datetime.datetime.now()
+        response = self._connect().create_security_group(GroupName=group_name,
+                                     Description=group_description,
+                                     VpcId=self.vpc_id)
+
+        self.security_group_ids = response['GroupId']
+        if self.verbose:
+            print('Temporary Security Group Created %s in vpc %s' % (self.security_group_ids, self.vpc_id))
+        data = self._connect().authorize_security_group_ingress(
+            GroupId=self.security_group_ids,
+            IpPermissions=[
+                {'IpProtocol': 'tcp',
+                 'FromPort': 22,
+                 'ToPort': 22,
+                 'IpRanges': [{'CidrIp': '0.0.0.0/0'}]}
+            ])
+
+        if self.verbose:
+            print("Successfully allowed incoming SSH port 22 for security group %s in %s" % (self.security_group_ids, self.vpc_id))
+
+    # ---------------------------------------------------------------------
+    def _create_vpc(self):
+        vpc_name = 'uploader-%s' % (random.randint(1,100))
+        response = self._connect().create_vpc(CidrBlock='192.168.0.0/16')
+        self.vpc_id = response['Vpc']['VpcId']
+        self._connect().create_tags(Resources=[self.vpc_id], Tags=[{'Key': 'Name', 'Value': vpc_name}])
+        if self.verbose:
+            print("Created VPC with id %s " % (self.vpc_id))
+
+    # ---------------------------------------------------------------------
+    def _create_vpc_subnet(self):
+        response = self._connect().create_subnet(CidrBlock='192.168.1.0/24', VpcId=self.vpc_id)
+        self.vpc_subnet_id = response['Subnet']['SubnetId']
+        self._connect().associate_route_table(SubnetId=self.vpc_subnet_id, RouteTableId=self.route_table_id)
+        self._connect().modify_subnet_attribute(MapPublicIpOnLaunch={ 'Value': True }, SubnetId=self.vpc_subnet_id)
+        if self.verbose:
+            print("Created VPC subnet with id %s" % (self.vpc_subnet_id))
+
+    # ---------------------------------------------------------------------
+    def prepare(self):
```

Comment:
drop this logic and incorporate it into ec2uploadimg command line processor. See previous comments